requirements/base.txt

```diff
@@ -6,3 +6,4 @@ igraph >=0.10.4
 optree >=0.11.0
 opt_einsum >= 3.3.0
 mpmath <1.4.0  # todo: teporarl pin for `NameError: name '_C' is not defined`
+dill >=0.3.8 # Support for 3.12
```

Comment:
You continue to insist on dill, I seem to not need it. Why is that?

Reply:
For some reason, it happens when `device='cuda'`. But not without it. If you re-run with cuda as the device do you run into the issue?

Reply:
I am using cuda. Can you try without dill and see if it works?
Thunder has a thing where it refuses to create cuda devices when there is no cuda (which is bad, arguably, but that's the status quo), but I can save and recover cuda as well.

Reply:
@t-vi I used your exact script that you posted on slack in a studio on an A10G. The one difference that gets it to fail is that I set `a, b = torch.randn(2, 5, device='cuda')`. If I `import dill as pickle` instead of importing normal pickle, everything works.